sklearn/datasets/_base.py

```diff
@@ -481,12 +545,9 @@ def load_iris(*, return_X_y=False, as_frame=False):
     >>> list(data.target_names)
     ['setosa', 'versicolor', 'virginica']
     """
-    module_path = dirname(__file__)
-    data, target, target_names = load_data(module_path, "iris.csv")
-    iris_csv_filename = join(module_path, "data", "iris.csv")
-
-    with open(join(module_path, "descr", "iris.rst")) as rst_file:
-        fdescr = rst_file.read()
+    data_file_name = "iris.csv"
+    data, target, target_names = load_csv_data(DATA_MODULE, data_file_name)
```

Comment:
I think we can avoid the extra indirection in a local variable that is used only once.

```suggestion
    data, target, target_names = load_csv_data(DATA_MODULE, "iris.csv")
```

Reply:
`data_file_name` is actually used again in constructing the `Bunch` below, which is why I used a variable. Do you still want me to make the change?